Print meaningful message when calling scripts in maven

res-life opened the pull request:
Print the error message when calling scripts in maven.
Add resultproperty and errorproperty to save the errorMsg and the exitCode, and then check.
This fixes #4292

Signed-off-by: Chong Gao <res_life@163.com>

res-life commented:
build

res-life commented:
@gerashegalov 
Because the Linux time command prints to stderr by default,  the error info includes time spent info. 


```
[ERROR] /home/chong/code/spark-rapids/dist/maven-antrun/build-parallel-worlds.xml:131: exec binary-dedupe.sh failed 
......
[ERROR] 
[ERROR] real	0m0.011s
[ERROR] user	0m0.008s
[ERROR] sys	0m0.005s
[ERROR] com/nvidia/spark/ExclusiveModeGpuDiscoveryPlugin.class is not bitwise-identical across shims
```

gerashegalov commented:
> @gerashegalov Because the Linux time command prints to stderr by default, the error info includes time spent info.
> 
> ```
> [ERROR] /home/chong/code/spark-rapids/dist/maven-antrun/build-parallel-worlds.xml:131: exec binary-dedupe.sh failed 
> ......
> [ERROR] 
> [ERROR] real	0m0.011s
> [ERROR] user	0m0.008s
> [ERROR] sys	0m0.005s
> [ERROR] com/nvidia/spark/ExclusiveModeGpuDiscoveryPlugin.class is not bitwise-identical across shims
> ```

I added these `time`s for diagnostics when we found that binary-dedupe was really slow on macOS but this is now mostly resolved and probably should not be logged to console. Feel free to either remove time wrappers or add `--append --output=<file>` to log timing info to a file.

res-life commented:
1. Removed time wrappers because the total time is printed.
```
[exec] binary-dedupe completed in 19 seconds
```
2. Added the exit code as part of the message.
3. Reordered the binary-dedupe.sh and check-shims-revisions.sh in the build-parallel-worlds.xml.

res-life commented:
build

gerashegalov commented on dist/maven-antrun/build-parallel-worlds.xml:
```diff
@@ -122,15 +122,34 @@
             <fileset dir="${project.build.directory}/parallel-world" includes="spark*/${shimServiceRsrc}"/>
         </concat>
 
+        <exec executable="${project.basedir}/scripts/binary-dedupe.sh"
```
What's the rationale of executing binary-dedupe.sh first? Previous iteration of the PR was what we wanted IMO. Now we allow again the build to fail on binary compatibility check whereas it's not the root cause in https://github.com/NVIDIA/spark-rapids/pull/4265#issuecomment-986072583

res-life replied:
Updated

res-life commented:
build

gerashegalov reviewed (approved):
LGTM

NvTimLiu reviewed (approved):
Looks good to me
